analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -64,24 +69,22 @@ private static ProgramState LearnBranchingConstraint<T>(ProgramState state, IBin
 
     private static SymbolicConstraint BinaryConstraint(BinaryOperatorKind kind, SymbolicValue left, SymbolicValue right)
     {
-        if (left is null && right is null)
-        {
-            return null;
```

Comment:
_Verification:_ Removing this short circuit has no functional change. It only deduplicates this `return null` with the `else return null`. And the new `?` down below are compensating for this.

Is that correct?

Reply:
Yes. That's right.